backend/product-listing.go

```diff
@@ -0,0 +1,116 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"fmt"
+	"html/template"
+	"net/http"
+	"path"
+	"strings"
+)
+
+const (
+	SAMPLE_TEMPLATE_FOLDER = "samples_templates"
+)
+
+type ProductPage struct {
+	Products     []Product
+	SearchAction string
+}
+
+type Product struct {
+	Name  string
+	Image string
+	URL   string
+	Stars string
+}
+
+var products []Product
+
+func InitProductListing() {
+	products = make([]Product, 0)
+	products = append(products,
+		Product{"Nexus 5x",
+			"https://lh3.googleusercontent.com/O00xttgXZHWYdpiWi7K0y7J2kvBMRXwlumBQNLV1853AMtf7MBBMSJ_ug9MDyUxMNjyo",
+			"https://store.google.com/product/nexus_5x",
+			"★★★★★"},
+		Product{"Nexus 5x Case",
+			"https://lh3.googleusercontent.com/O00xttgXZHWYdpiWi7K0y7J2kvBMRXwlumBQNLV1853AMtf7MBBMSJ_ug9MDyUxMNjyo",
+			"https://store.google.com/product/adopted_case_n5x",
+			"★★★★★"},
+		Product{"Nexus 5x Folio",
+			"https://lh3.googleusercontent.com/ttVsejKSBY5OLLDj38d1YddiBjkvK9BXqIOzIwa7thdWpMoAJzyhdfYixsl0G2Rx_W8",
+			"https://store.google.com/product/adopted_folio_n6p",
+			"★"},
+		Product{"Nexus 6P",
+			"https://lh3.googleusercontent.com/meAtjplzh2B6G9n8kipX7vZ9cti4cUlk48ggRsWiCge1_b_6Ni2U9PFEs6l0RQSlvqGH",
+			"https://store.google.com/product/nexus_6p",
+			"★★★★"},
+		Product{"Speck CandyShell Grip Case for Nexus 6P",
+			"https://lh3.googleusercontent.com/a_JFyb2hYBMd-NVaGZWq2piD6txa6cG3DV2KI2i3k59HRuYDZ8lXYe5m9zrACtFFIA_K",
+			"https://store.google.com/product/speck_candyshell_grip_case_n6p",
+			"★★"},
+		Product{"Chromecast Audio",
+			"https://lh3.googleusercontent.com/ysmhjWszkLsyiWRJ97JZyL0oXL0HUzKXfhOgYC8wvLHhQUwr_dRADE1tBvUNzXPGrA",
+			"https://store.google.com/product/chromecast_audio",
+			"★★★★★"},
+	)
+	registerProductListingHandler("product_listing")
+	registerProductListingHandler("product_listing/preview")
+}
+
+func registerProductListingHandler(sampleName string) {
+	route := path.Join("/", SAMPLE_TEMPLATE_FOLDER, sampleName) + "/"
```

Comment:
Line 87 (below the code) I am using that constant and having the `/` will make things difficult.

`filePath := path.Join(DIST_FOLDER, SAMPLE_TEMPLATE_FOLDER, sampleName, "index.html")`